Zend/zend_API.c

```diff
@@ -4567,7 +4567,12 @@ ZEND_API zend_property_info *zend_declare_typed_property(zend_class_entry *ce, z
 	 * anyway, and assert after inheritance that the property is not actually virtual. */
 	if (access_type & ZEND_ACC_VIRTUAL) {
 		if (Z_TYPE_P(property) == IS_UNDEF) {
-			property_info->offset = (uint32_t)-1;
+			property_info->offset = OBJ_PROP_TO_OFFSET(ce->default_properties_count);
```

Comment:
Doesn't this break the `@ virtual` annotation, and thus reintroduces https://github.com/php/php-src/issues/13988 ?

Reply:
The offset is later removed again...

https://github.com/php/php-src/pull/17870/files#diff-16d00017a3ed574cca152e8754246a1424db74e221934eb1c0c2513442882615R1697

In any case, I think it's better to rely on `ZEND_ACC_VIRTUAL` when possible.

Reply:
Ah okay I missed this.